vllm/entrypoints/openai/serving_engine.py

```diff
@@ -47,24 +47,9 @@ def __init__(self,
             ]
 
         self.max_model_len = 0
-        # Lazy initialized
         self.tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast]
 
-        try:
-            event_loop = asyncio.get_running_loop()
-        except RuntimeError:
-            event_loop = None
-
-        if event_loop is not None and event_loop.is_running():
-            # If the current is instanced by Ray Serve,
-            # there is already a running event loop
-            event_loop.create_task(self._post_init())
-        else:
-            # When using single vLLM without engine_use_ray
-            asyncio.run(self._post_init())
-
-    async def _post_init(self):
-        engine_model_config = await self.engine.get_model_config()
+        engine_model_config = asyncio.run(self.engine.get_model_config())
```

Comment:
I think the main problem was stated in the comments before. The init method might be already running inside an asyncio loop, which makes it not possible to call `asyncio.run` unless `nest_asyncio` is used.

Reply:
Gotcha - I misunderstood the issue, I think we can just use run_until_complete with the event loop